app/src/main/java/org/simple/clinic/storage/DaoWithUpsert.kt

```diff
@@ -0,0 +1,34 @@
+package org.simple.clinic.storage
+
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy
+import android.arch.persistence.room.Transaction
+import android.arch.persistence.room.Update
+
+/**
+ * Mimics sqlite's `upsert` conflict strategy because Room's grammar doesn't understand it.
+ */
+abstract class DaoWithUpsert<T> {
```

Comment:
@sanchita @vinaysshenoy suggestions for a better name? `UpsertableDao`? `DaoWithUpsertClause`? or is this just fine?

Reply:
I think it's fine.